### ISSUE TITLE
Uncaught exception thrown from 'clientWaitAsync' function

### ISSUE THREAD

**Issue by onderilkesever** [ThatOpen/engine_components#483]
### Describe the bug 📝

I am using MeshCullerRenderer, and after I am done with it, I dispose it. However, sometimes `Uncaught (in promise) undefined` errors appear on console. After detailed investigation, I think I have an idea what is causing it. When disposing, WEBGL context is being killed. However, if there was already `readPixelsAsync` call scheduled from previous interval. It goes through `getBufferSubDataAsync` > `clientWaitAsync` > `clientWaitSync` and it hits below reject statement
```
      const res = gl.clientWaitSync(sync, flags, 0);
      if (res === gl.WAIT_FAILED) {
        reject();
        return;
      }
```
And this exception is never catched, thus, arrives our application in the end. In order to avoid this exception unnecessarily leaking to our app, I believe it needs to be catched somewhere or instead of rejecting silently return might be an option. 
Can you please have a look and let me know if I am missing something. 
Thanks in advance.

### System Info 💻

```shell
System:
    OS: macOS 14.6.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 79.83 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.19.0 - ~/.nvm/versions/node/v18.19.0/bin/node
    npm: 10.2.3 - ~/.nvm/versions/node/v18.19.0/bin/npm
  Browsers:
    Chrome: 127.0.6533.120
    Edge: 127.0.2651.105
    Firefox: 123.0.1
    Safari: 17.6
```

### Used Package Manager 📦

npm

### Validations ✅

- [X] Read the [docs](https://docs.thatopen.com/intro).
- [X] Check that there isn't [already an issue](https://github.com/ThatOpen/engine_components/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a repository issue and not a framework-specific issue. For example, if it's a THREE.js related bug, it should likely be reported to [mrdoob/threejs](https://github.com/mrdoob/three.js) instead.
- [X] Check that this is a concrete bug. For Q&A join our [Community](https://people.thatopen.com/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by agviegas** [comment]
Hey @onderilkesever so you would like us to remove the `reject()` line, right?

**Comment by agviegas** [comment]
@onderilkesever fixed in `2.2.8`

**Comment by onderilkesever** [comment]
thanks for prompt action here @agviegas
